### ISSUE TITLE
Batchwise outer product?

### ISSUE THREAD

**Issue by maximus009** [fchollet/keras#5418]
I see that the backend has an implementation for batchwise dot product.
Can there be a function to do batchwise outer product?

I tried implementing an outer product Layer of my own, but I am getting stuck (dealing with batches) Would appreciate some pointers or help. 

Thanks in advance.

**Comment by maximus009** [comment]
Basically, all I want to do is take two tensors (coming from different layers) and return the outer product.
dim(?, 1, d) x dim(?, d, 1) = dim(?, d, d).
where, '?' represents batch size.

I'm sure this is trivial, but I'm unable to wrap my head around it.

**Comment by unrealwill** [comment]
Hello,

Have you tried looking at https://keras.io/backend/#backend-functions : batch_dot 
You probably need a few reshapes before but it should do it.

**Comment by maximus009** [comment]
Yep, I was implementing something along those lines. Thank you for your quick response.
Managed to implement using a Lambda layer.

You can find the code here:
https://gist.github.com/maximus009/6d8a2e4d6d56d02e5581eb11ab1b2ea2

**Comment by adelsalehali1982** [comment]
Hi @ maximus009  could you add an example how to use your code
